kahuna/public/js/components/gr-chips/gr-chips.js

```diff
@@ -0,0 +1,241 @@
+import angular from 'angular';
+
+import './gr-chips.css!';
+import template from './gr-chips.html!text';
+
+import {grTextChip}          from './gr-text-chip';
+import {grFilterChip}        from './gr-filter-chip';
+import {grStaticFilterChip}  from './gr-static-filter-chip';
+import {grFilterChooserChip} from './gr-filter-chooser-chip';
+
+
+export const grChips = angular.module('gr.chips', [
+    grTextChip.name,
+    grFilterChip.name,
+    grStaticFilterChip.name,
+    grFilterChooserChip.name
+]);
+
+grChips.controller('grChipsCtrl', ['$scope', function($scope) {
+    const $grChipsCtrl = this;
```

Comment:
If you mean the `$` prefix, I borrowed this from angular-material coding style as I thought their code looked [nicely organised](https://github.com/angular/material/blob/master/src/components/chips/js/chipsDirective.js) and that it made the reference a little clearer in the templates and elsewhere. I'm not obsessed about it particularly so could remove the `$` if people prefer.

If you mean naming each controller differently rather than `ctrl`, this is to make things more self-explanatory when directives manipulate 2-3 different other controllers than themselves. It's also used to access the reference for the "parent" directive (e.g. `grChipsCtrl` from within a `grTextChip`) within the template, through scope inheritance, with hierarchy enforced via `require`.